pkg/scanner/scanner.go

```diff
@@ -0,0 +1,170 @@
+package scanner
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"path/filepath"
+	"strings"
+
+	"github.com/fluxcd/pkg/version"
+	sourcev1 "github.com/fluxcd/source-controller/api/v1beta1"
+	"github.com/go-logr/logr"
+	profilesv1 "github.com/weaveworks/profiles/api/v1alpha1"
+	"github.com/weaveworks/profiles/pkg/gitrepository"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/yaml"
+)
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -generate
+//counterfeiter:generate -o fakes/fake_git_client.go . GitClient
+//GitClient client for interacting with git
+type GitClient interface {
+	ListTags(url string, secret *corev1.Secret) ([]string, error)
+}
+
+//counterfeiter:generate -o fakes/fake_repo_manager.go . GitRepositoryManager
+//GitRepositoryManager for managing gitrepositorys
+type GitRepositoryManager interface {
+	CreateAndWaitForResources(repo profilesv1.Repository, tags []gitrepository.Instance) ([]*sourcev1.GitRepository, error)
+	DeleteResources([]*sourcev1.GitRepository) error
+}
+
+//counterfeiter:generate -o fakes/fake_http_client.go . HTTPClient
+//HTTPClient for making HTTP requests
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+//Scanner for scanning repositorys
+type Scanner struct {
+	gitRepositoryManager GitRepositoryManager
+	gitClient            GitClient
+	httpClient           HTTPClient
+	logger               logr.Logger
+	namespace            string
+	ctx                  context.Context
+}
+
+//New returns a Scanner
+func New(gitRepositoryManager GitRepositoryManager, gitClient GitClient, httpClient HTTPClient, logger logr.Logger) *Scanner {
```

Comment:
Should we create a config struct? do you think this will expand?
I guess it's fine, but if we add one more we should. :D

Reply:
I think its okay. `httpClient` and `gitCleint` are only exposed for testing. I could switch to the `export_test` method instead to reduce this to 2 arguments